test/src/test/java/org/corfudb/runtime/collections/CorfuStoreShimTest.java

```diff
@@ -862,6 +863,171 @@ public void testNestedSecondaryIndexesRepeatedFieldFollowedByNonPrimitive() thro
         }
     }
 
+    /**
+     * Example to see how nested secondary indexes work when a secondary index is set on a 'oneOf' field.
+     * This test combines all possibilities of secondary indexes on 'oneOf' properties:
+     * (1) 'oneOf' index nested in a repeated field
+     *      (a) Index primitive type
+     *      (b) Index non-Primitive type
+     *      (c) Index repeated Primitive
+     * (2) 'oneOf' not-nested in a repeated field
+     *
+     * Please see example_schemas.proto: ContactBook
+     *
+     * In this test, we create 3 contact books: work, friends & family
+     * A 'contact book' is composed of a list of 'contact', each contact has a 'name' and its 'contactInformation'
+     * is either an 'address' (which has a field city) or a 'phone number' (contactInformation is 'oneOf')
+     * A 'common' phone number, is a number shared across different contacts
+     *
+     * * ContactBookWork (4 contacts):
+     *   - 1 with San Francisco Address
+     *   - 3 with Phone Number: 1 contact with common phone number (home and mobile)
+     *                          1 work contact with unique phone number
+     *                          1 contact with a common mobile number (its own home number)
```

Comment:
what is the difference between this and L884(common number home and mobile)?

Reply:
This contact has a different home number, but a common mobile with another contact... rather L.884 has exactly the same phone number (home/mobiles)